services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/ProjectSettingsManagerImpl.java

```diff
@@ -194,10 +196,12 @@ public void deleteProjectSetting(UserInfo userInfo, String id) throws DatastoreE
 			validateGoogleCloudBucketOwnership(externalGoogleCloudStorageLocationSetting, getBucketOwnerAliases(userInfo.getId()));
 		} else if (storageLocationSetting instanceof ExternalStorageLocationSetting) {
 			ExternalStorageLocationSetting externalStorageLocationSetting = (ExternalStorageLocationSetting) storageLocationSetting;
+			ValidateArgument.required(externalStorageLocationSetting.getUploadType(), "uploadType");
 			ValidateArgument.required(externalStorageLocationSetting.getUrl(), "url");
 			ValidateArgument.validExternalUrl(externalStorageLocationSetting.getUrl());
 		} else if (storageLocationSetting instanceof ExternalObjectStorageLocationSetting) {
 			ExternalObjectStorageLocationSetting externalObjectStorageLocationSetting = (ExternalObjectStorageLocationSetting) storageLocationSetting;
+			ValidateArgument.required(externalObjectStorageLocationSetting.getUploadType(), "uploadType");
```

Comment:
Move these ValidateArgument calls to after the if-else blocks, so we catch this for any new Storage Location types.